apps/meteor/app/ui-message/client/ActionManager.ts

```diff
@@ -71,34 +71,41 @@ export class ActionManager implements IActionManager {
 		return triggerId;
 	}
 
-	public async emitInteraction(appId: string, userInteraction: DistributiveOmit<UiKit.UserInteraction, 'triggerId'>) {
+	public async emitInteraction(
+		appId: string,
+		userInteraction: DistributiveOmit<UiKit.UserInteraction, 'triggerId'>,
+	): Promise<Pick<UiKit.ServerInteraction, 'type'> | undefined> {
 		this.notifyBusy();
 
 		const triggerId = this.generateTriggerId(appId);
-
 		let timeout: ReturnType<typeof setTimeout> | undefined;
 
-		await Promise.race([
-			new Promise((_, reject) => {
+		try {
+			void new Promise((_, reject) => {
 				timeout = setTimeout(() => reject(new UiKitTriggerTimeoutError('Timeout', { triggerId, appId })), ActionManager.TRIGGER_TIMEOUT);
-			}),
-			sdk.rest
-				.post(`/apps/ui.interaction/${appId}`, {
-					...userInteraction,
-					triggerId,
-				})
-				.then((interaction) => this.handleServerInteraction(interaction)),
-		]).finally(() => {
+			});
+
+			const interaction = await sdk.rest.post(`/apps/ui.interaction/${appId}`, {
+				...userInteraction,
+				triggerId,
+			});
+
+			return this.handleServerInteraction(interaction);
+		} finally {
```

Comment:
I'm unsure if this new implementation has the same behavior as the old one. The previous implementation used `Promise.race` to add a manual timeout to the HTTP call, with the new version it seems the `void` makes the promise result ignored and also makes the code wait for the HTTP call to be fulfilled, which in the previous version this was not true, the method finishes the execution when the first promise is fulfilled, ignoring completely the other one.

- What happens if the `reject`ion happens indeed? Would the method still throw an error to the caller function?
- Will the error being thrown to the upper layers and when the HTTP call finishes, also make the value be returned (since now we expect a value from the method)?
- Is try/finally block identical to what Promise.race/finally does speaking about the error being thrown to the upper layers?

Finally, since even the previous implementation adds some complexity and unpredictability to it, can we add a unit test (could be just for this method) just to make sure the method is doing what we expect?